axelrod/strategy_transformers.py

```diff
@@ -228,11 +297,21 @@ def dual_wrapper(player, opponent, proposed_action):
     -------
     action: an axelrod.Action, C or D
     """
+
     if not player.history:
-        player.original_player = player.original_class(**player.init_kwargs)
+        player.use_history = []
+
+    temp = player.history[:]
+    player.history = player.use_history[:]
+
```

Comment:
> To ensure I'm following: step 5 and 6 could be swapped? (Not suggesting they are, just checking I'm following).

yes, that is correct.

> and step 3 is the important one right? Step 3 is the step that's ensuring the state is updated "properly"

yes.  step 3 automatically happens as a result of step 2. if it didn't, this would all fall apart.